libcontainer/container_linux.go

```diff
@@ -512,6 +512,43 @@ func (c *linuxContainer) commandTemplate(p *Process, childInitPipe *os.File, chi
 	return cmd
 }
 
+// shouldSendMountSources says whether the child process must setup bind mounts with
+// the source pre-opened (O_PATH) in the host user namespace.
+// See https://github.com/opencontainers/runc/issues/2484
+func (c *linuxContainer) shouldSendMountSources() bool {
+	// Passing the mount sources via SCM_RIGHTS is only necessary when
+	// both userns and mntns are active.
+	if !c.config.Namespaces.Contains(configs.NEWUSER) ||
+		!c.config.Namespaces.Contains(configs.NEWNS) {
+		return false
+	}
+
+	// We don't need to send sources if there are no bind-mounts.
+	if len(c.config.Mounts) == 0 {
+		return false
+	}
```

Comment:
Looks like this check is redundant, and won't save any resources either, as in case `len(c.config.Mounts) == 0` the loop below will have no iterations and we'll return after checking `!bindMounts`.

Reply:
heh, right. Thanks!